### PR TITLE
DataViews: Mark the new Templates pages as stable

### PR DESCRIPTION
**Pull request by youknowriad**
Related #55083 

 - This PR stabilizes the new templates page based on data views as table and replaces entirely the old page. 
 - It keeps the "view" menu hidden behind the experimental flag because the "side by side" and "grid views" are not ready yet.
 - It would be good to replace the template parts page as well (which will allow us to remove a bunch of other components).

The idea here is to start getting some real usage feedback about the dataviews, to be completely ready when we reach the beta1 for the new WP release.

**Testing instructions**

 - Disabled the "data views" experiment.
 - Open the "browse all templates" page in the site editor.
 - It should show the new data views page without the "view" menu.

### CODE REVIEW

**Comment by youknowriad** [comment]
I noticed that the existing page has a "Add template" button at the top, not sure if want to add that button first before merging this one.

File: packages/edit-site/src/components/page-templates/index.js [hunk 1]
```diff
@@ -1,92 +1,407 @@
+/**
```
Comment (youknowriad) [inline packages/edit-site/src/components/page-templates/index.js:1]: This is just the old dataviews-template.js without any change.

File: packages/dataviews/src/dataviews.js [hunk 2]
```diff
@@ -68,12 +68,14 @@ export default function DataViews( {
 							onChangeView={ onChangeView }
 						/>
 					</HStack>
-					<ViewActions
-						fields={ fields }
-						view={ view }
-						onChangeView={ onChangeView }
-						supportedLayouts={ supportedLayouts }
-					/>
+					{ window?.__experimentalAdminViews && (
```
Comment (ntsekouras) [inline packages/dataviews/src/dataviews.js]: We should probably just hide the `ViewTypeMenu`, no?

Comment (youknowriad) [reply packages/dataviews/src/dataviews.js]: I'm open to that.

**Review by ntsekouras — APPROVED** [review]
Thanks Riad!

>existing page has a "Add template" button at the top

We should probably add this button too and let's ship.

**Comment by mcsf** [comment]
Nice, that was fast. :)

**Pull request merged**